README.md

```diff
@@ -1,368 +1,276 @@
-A stage can be a string specifying the stage name or an object specifying the stage name and any additional parameters.
-Stages are executed in the order listed.
+Execute a sequence of operations that are described in a JSON file.
 
-This example `pipeline.json` gzips the input tilest and saves it in the given output directory.
+> **Note:** The pipeline execution feature is preliminary. Many aspects of the pipeline definition, including the JSON representation and the exact set of operations that are supported as parts of pipelines may change in future releases.
 
-```json
-{
-    "input": "Tileset/",
-    "output": "TilesetGzipped/",
-    "stages": ["gzip"]
-}
-```
+ The basic structure of a pipeline JSON file is summarized here:
 
-This pipeline uncompresses the input tileset and then compresses all the tiles. Files like tileset.json are left uncompressed.
+- A pipeline has an `input` and `output`, which are the names of a tileset directory or package
+- A pipeline has an array of 'tileset stages'
+- A tileset stage has a `name` and a `description`
+- A tileset stage has an array of 'content stages'
+- A content stage has a `name` and a `description`
+- A content stage can carry information about the content types that it is applied to
 
-```json
+A simple example pipline may therefore look like this:
+```
 {
-    "input": "Tileset/",
-    "output": "TilesetGzipped/",
-    "stages": [
-        "ungzip",
+  "input": "./specs/data/TilesetOfTilesetsWithUris",
+  "output": "./output/TilesetOfTilesetsWithUris.3tz",
+  "tilesetStages": [
+    {
+      "name": "_b3dmToGlb",
+      "description": "Convert B3DM to GLB",
+      "contentStages": [
         {
-            "name": "gzip",
-            "tilesOnly": true
+          "name": "b3dmToGlb",
+          "description": "Convert each B3DM content into GLB"
         }
-    ]
+      ]
+    }
+  ]
 }
 ```
 
-###Pipeline Stages
+The `name` of a tileset- or content stage can refer to a predefined set of operations that can be executed. If a `name` is not one of the known operations, it should start with an `_` underscore. 
 
-####gzip
+The `description` of a tileset- or content stage is intended as a human-readable summary, to be shown as log output.
 
-Gzips the input tileset.
+The predefined operations largely correspond to the command-line functionality. 
 
-**Properties**
+The known tileset stages are:
 
-|   |Type|Description|Required|
-|---|----|-----------|--------|
-|**tilesOnly**|`boolean`|Only gzip tiles.|No, default: `false`|
+- `upgrade`: Upgrade the input tileset to the latest version. Details about what that means are omitted here.
+- `combine`: Combine all external tilesets of the input tileset, to create a single tileset
 
-####ungzip
+The known content stages are:
 
-Ungzips the input tileset.
+- Compression:
+  - `gzip`: Apply GZIP compression to all files (with optional filters)
+  - `ungzip`: Uncompress all files that are compressed with GZIP
```

Comment:
Do gzip/ungzip also apply to the root tileset.json? Since they're content stages I wasn't sure.

Reply:
Yes. We could try to find a better term than "content" here. It's all files/entries of a tileset data set.

Reply:
Well, this is embarassing... :-( https://github.com/CesiumGS/3d-tiles-tools/issues/22

I'm currently trying to see whether it makes sense to convert them into tileset stages. Not only would that avoid confusion, it would also resolve a bunch of questions about the special handling of the `tileset.json` when they are supposed to be a "content stage"...